ts/src/huobi.ts

```diff
@@ -2809,120 +2898,126 @@ export default class huobi extends Exchange {
          * @param {object} [params] extra parameters specific to the huobi api endpoint
          * @returns {object} an associative dictionary of currencies
          */
-        const response = await this.spotPublicGetV2ReferenceCurrencies (params);
+        const response = await this.spotPublicGetV1SettingsCommonChains (params);
```

Comment:
@ttodua why is this endpoint better than the other one? THe other one returned the chains indexed by currency, which seems to make more sense in this context

Reply:
because that other (old) endpoint removed any indication of chains. see here:
- https://api.huobi.pro/v2/reference/currencies
take the first item right away, which says chain-id & chain name `algousdt`.  but in that object there is no indication of which chain is that actually (there should be something separate name, like `algo` or `algorand` or whatever the network indentifier). `algousdt` is no network identifier, because there are also `algousdc` etc... 

on the contrary:
- https://api.huobi.pro/v1/settings/common/chains
as link also includes `chains`, this endpoint is more suited for networks. just search `algousdt` and you will find an object which has more details network information

Reply:
![image](https://github.com/ccxt/ccxt/assets/43336371/a362a176-785f-425e-bebf-7c1191b1251e)
![image](https://github.com/ccxt/ccxt/assets/43336371/5fbaab68-9f2b-4c5a-ba0b-193ec2af2fba)


@ttodua sorry, I'm not clear yet, comparing both versions, which information is missing in v2?

Reply:
> @ttodua sorry, I'm not clear yet, comparing both versions, which information is missing in v2?

in my above comment I've mentioned a first straightforward example (algousdt) you would come across. also named few more examples in another comment, and can add more other examples too if you want

Reply:
no, you can't just replace `usdt` with empty space, because the logic is not about `usdt` currency specifically, but all currencies out there (on huobi) have random assigned chain-ids, where there are no involvement of currency names and nothing can be replaced. for example, for `XYZ` coin the chain-id for tron might be something like `tr931` or alike random ids where there is not predefined rule to link one of them. we shouldn't do any manipulations to guess network names, so that's why the one-to-one mapping is needed to have 100% correct approach.

Reply:
> might be deprecared soon
there is no announcement for that endpoint deprecation even though it's v1 (and from this standpoint I doubt they would deprecate it in near future, as it's the only way atm to use API to really withdraw/deposit coins).
and if they will ever do, ofc, we are here to maintain the lib.

Reply:
@ttodua can you find an example of a chain that does not have this `baseChain` field and this replacement wouldn't work? (just to make sure)
![image](https://github.com/ccxt/ccxt/assets/43336371/98b875fd-101f-472a-8eb5-4bf4544eee7b)

Reply:
to keep everything reliable, as you know, we need network tests (which is impossible without networks-unifications from base and then on individual exchanges)

Reply:
> baseChain

if you search for `"chain":` in that json, there are 1064 cases, but only 881 entries have `baseChain`, so it is not reliable.

Reply:
@ttodua ok let's keep the v1 for now